doc/languages-frameworks/typst.section.md

```diff
@@ -0,0 +1,55 @@
+# Typst {#typst}
+
+Typst can be configured to include packages from Typst Universe or customized packages.
+
+## Custom Environment {#typst-custom-environment}
+
+Creating a custom Typst environment with a selected set of packages from Typst Universe can be done with the following code. It is possible to specify a specific version of a Typst package (e.g., `cetz_0_3_0`). A package without a version number refers to its latest version.
+
+```nix
+typst.withPackages (p: with p; [
+  polylux
+  cetz_0_3_0
+])
+```
+
+Since Typst Universe doesn't provide an way to fetch a package with a specific hash, it can be common that Typst packages' hash in nixpkgs are outdated. The example below can fix this issue.
```

Comment:
```suggestion
Since Typst Universe doesn't provide a way to fetch a package with a specific hash, it can be common that Typst packages' hash in nixpkgs are outdated. The example below can fix this issue.
```